### PR TITLE
Remove hard coded runtime

### PR DESCRIPTION
**Pull request by micholas**
## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Infrastructure (change which improves the lifecycle management (CI/CD, build, package, deploy, lint, etc) of the application, but does not change functionality.)
- [ ] Technical debt (change which improves the maintainability of the codebase, but does not change functionality)
- [ ] Testing (change which modifies or adds test coverage, but does not change functionality)
- [ ] Documentation (change which modifies or adds documentation, but does not change functionality)

## Description

Determine runtime from SAM template instead of using hard coded value

## Testing

Created unit tests for extracting runtime from SAM template

## Checklist

- [x] I have read the **README** document
- [x] I have read the **CONTRIBUTING** document
- [x] My code follows the code style of this project
- [x] I have added tests to cover my changes
- [x] All new and existing tests passed
- [ ] A short description of the change has been added to the **CHANGELOG**

## License

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by bryceitoc9** [comment]
Can we have some sort of way to select/add a default runtime? This breaks BYOC (assuming we want to keep that around).

**Comment by micholas** [comment]
@bryceitoc9 We have decided to use SAM as a basis for this extension. The 1st step when creating a SAM project is to to [initialize](https://docs.aws.amazon.com/serverless-application-model/latest/developerguide/sam-cli-command-reference-sam-init.html) it with `sam init`. In my opinion if a user has not yet initialized their sam project, informing/assisting them to initialize SAM is our top priority and arguably the only thing we should do for them. If they haven't run sam init, it's not a sam project.

If we are to forgo the SAM requirement, I'd recommend leveraging serverless. The last thing I'd recommend is to attempt to re-invent SAM or the Serverless framework. With either of those step zero is initialize. Those project don't attempt to guess the runtime and the safest bet is for us not to attempt to do so either.

File: src/shared/codelens/localLambdaRunner.ts [hunk 1]
```diff
@@ -323,6 +323,47 @@ export class LocalLambdaRunner {
...
+    templatePath: string,
+}): Promise<string> {
+    const samTemplateData: CloudFormation.Template = await CloudFormation.load(params.templatePath)
+    if (!samTemplateData.Resources) {
+        throw new Error(
+            `Please specify Resource for '${params.handlerName}' Lambda in SAM template: '${params.templatePath}'`
+        )
+    }
+    const runtimes = new Set<string>()
+    for (const resourceKey in samTemplateData.Resources) {
```
Comment (awschristou) [inline src/shared/codelens/localLambdaRunner.ts]: Is `for (const resourceKey in samTemplateData.Resources)` functionally equivalent to working with `Object.getOwnPropertyNames(samTemplateData.Resources)`? This is how detectLocalLambdas.ts and a few other places seem to process the `Resources` field.

Comment (micholas) [reply src/shared/codelens/localLambdaRunner.ts]: [for in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) and [for of](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...of) iterate over the enumerable properties whereas [Object.getOwnPropertyNames](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/getOwnPropertyNames) 

> returns an array of all properties (including non-enumerable properties except for those which use Symbol) found directly in a given object

In this case the result is likely the same, however; generally one does not expect to iterate over non-enumerable properties.

**Comment by awschristou** [comment]
This PR applies to both #432 and #433

**Pull request merged**